cmake/KokkosConfig.cmake.in

```diff
@@ -28,45 +38,37 @@ if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS AND "separable_compilation"
   )
   message(FATAL_ERROR "Conflicting COMPONENTS: 'launch_compiler' and 'separable_compilation'")
 endif()
+# end of components deprecation
 
-if("launch_compiler" IN_LIST Kokkos_FIND_COMPONENTS)
-  #
-  # if find_package(Kokkos COMPONENTS launch_compiler) then rely on the
-  # RULE_LAUNCH_COMPILE and RULE_LAUNCH_LINK to always redirect to the
-  # appropriate compiler for Kokkos
-  #
-
-  message(
-    STATUS
-      "kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos"
-  )
-  kokkos_compilation(GLOBAL CHECK_CUDA_COMPILES)
+if(NOT DEFINED Kokkos_GLOBAL_COMPILATION)
+  set(Kokkos_GLOBAL_COMPILATION true)
+endif()
 
-elseif(@Kokkos_ENABLE_CUDA@ AND NOT @KOKKOS_COMPILE_LANGUAGE@ STREQUAL CUDA AND NOT "separable_compilation" IN_LIST
-                                                                                Kokkos_FIND_COMPONENTS
-)
-  #
-  # if CUDA was enabled, the compilation language was not set to CUDA, and separable compilation was not
-  # specified, then set the RULE_LAUNCH_COMPILE and RULE_LAUNCH_LINK globally and
-  # kokkos_launch_compiler will re-direct to the compiler used to compile CUDA code during installation.
-  # kokkos_launch_compiler will re-direct if ${CMAKE_CXX_COMPILER} and -DKOKKOS_DEPENDENCE is present,
-  # otherwise, the original command will be executed
-  #
+if(NOT DEFINED Kokkos_FORCE_COMPILER_LAUNCHER)
+  set(Kokkos_FORCE_COMPILER_LAUNCHER false)
+endif()
 
-  # run test to see if CMAKE_CXX_COMPILER=nvcc_wrapper
-  kokkos_compiler_is_nvcc(IS_NVCC ${CMAKE_CXX_COMPILER})
+if(Kokkos_GLOBAL_COMPILATION)
+  # If nvcc is not the CXX_COMPILER we go through the launcher
+  if(@Kokkos_ENABLE_CUDA@ AND NOT @KOKKOS_COMPILE_LANGUAGE@ STREQUAL CUDA)
+    kokkos_compiler_is_nvcc(IS_NVCC ${CMAKE_CXX_COMPILER})
+    if(NOT IS_NVCC)
+      message(
+        STATUS
+          "kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos"
+      )
+      kokkos_compilation(GLOBAL)
+    endif()
+  endif()
 
-  # if not nvcc_wrapper and Kokkos_LAUNCH_COMPILER was not set to OFF
-  if(NOT IS_NVCC AND (NOT DEFINED Kokkos_LAUNCH_COMPILER OR Kokkos_LAUNCH_COMPILER))
+  # If users request the compiler launcher we use it globally
+  if(Kokkos_FORCE_COMPILER_LAUNCHER)
     message(
       STATUS
         "kokkos_launch_compiler is enabled globally. C++ compiler commands with -DKOKKOS_DEPENDENCE will be redirected to the appropriate compiler for Kokkos"
     )
     kokkos_compilation(GLOBAL)
   endif()
-
-  # be mindful of the environment, pollution is bad
-  unset(IS_NVCC)
```

Comment:
Did you mean to remove these lines?

Reply:
readded in https://github.com/kokkos/kokkos/pull/7581/commits/80becabf9012bced09c4a45f7088616a5bc37e41 but without the useless comment